File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/dummy/QuestionarioDummyDataBase.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao.dummy;

import net.fbvictorhugo.j.barreirasanitaria.data.dao.IQuestionarioDAO;
import net.fbvictorhugo.j.barreirasanitaria.data.model.Questionario;

import java.util.ArrayList;
import java.util.List;

public class QuestionarioDummyDataBase implements IQuestionarioDAO {

    private static QuestionarioDummyDataBase sFactory;
    private List<Questionario> mDummyList;

    public static QuestionarioDummyDataBase getInstance() {
        if (sFactory == null) {
            sFactory = new QuestionarioDummyDataBase();
        }
        return sFactory;
    }

    private QuestionarioDummyDataBase() {
        mDummyList = new ArrayList<>();
    }

    @Override
    public List<Questionario> listar() {
        return mDummyList;
    }

    @Override
    public void inserir(Questionario questionario) throws Exception {
        //TODO("Not yet implemented")
    }

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/dummy/QuestionarioDummyDataBase.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao.dummy

import net.fbvictorhugo.k.barreirasanitaria.data.dao.IQuestionarioDAO
import net.fbvictorhugo.k.barreirasanitaria.data.model.Questionario

object QuestionarioDummyDataBase : IQuestionarioDAO {

    private val mDummyList: ArrayList<Questionario> = ArrayList()

    override fun listar(): ArrayList<Questionario> {
        return mDummyList
    }

    override fun inserir(questionario: Questionario) {
        TODO("Not yet implemented")
    }

}
